File: backend/app/schemas/review_queue.py
from pydantic import BaseModel


class ReviewQueueItem(BaseModel):
    review_item_id: int
    finding_id: int
    analysis_id: str
    application_name: str
    finding_headline: str
    kb_reference: str
    review_reason: str
    assigned_owner_user_id: str
    due_date: str
    review_status: str
    created_utc: int
    updated_utc: int
    resolution_note: str | None = None
    defer_reason: str | None = None
    is_overdue: bool


class ReviewQueueResponse(BaseModel):
    items: list[ReviewQueueItem]

File: frontend/src/services/reviewQueue.ts
import { apiGet } from "./api";

export type ReviewQueueItem = {
  review_item_id: number;
  finding_id: number;
  analysis_id: string;
  application_name: string;
  finding_headline: string;
  kb_reference: string;
  review_reason: string;
  assigned_owner_user_id: string;
  due_date: string;
  review_status: string;
  created_utc: number;
  updated_utc: number;
  resolution_note: string | null;
  defer_reason: string | null;
};

export type ReviewQueueResponse = {
  items: ReviewQueueItem[];
};

export function getReviewQueue() {
  return apiGet<ReviewQueueResponse>("/review-queue");
}